app/models/user.rb

```diff
@@ -35,15 +42,25 @@ def new_token
 
   def remember
     self.remember_token = User.new_token
-    update_attribute :remember_digest, User.digest(remember_token)
+    update_attribute :remember_digest, User.digest remember_token
   end
 
-  def authenticated? remember_token
+  def authenticated? arttribute, token
+    digest = send "#{arttribute}_digest"
     return false if remember_digest.nil?
     BCrypt::Password.new(remember_digest).is_password? remember_token
   end
 
   def forget
     update_attribute :remember_digest, nil
   end
+
+  def activate
+    update_attribute :activated, true
+    update_attribute :activated_at, Time.zone.now
```

Comment:
you can use update_attributes
uodate_attributes activated: true, activated_at: Time.zone.now

Reply:
done